submissions/api.py

```diff
@@ -201,12 +203,18 @@ def _get_submission_model(uuid, read_replica=False):
     Helper to retrieve a given Submission object from the database. Helper is needed to centralize logic that fixes
     EDUCATOR-1090, because uuids are stored both with and without hyphens.
     """
+    from nose.tools import set_trace; set_trace()
     submission_qs = Submission.objects
+    old_submission_qs = HyphenatedUUIDSubmission.objects
     if read_replica:
         submission_qs = _use_read_replica(submission_qs)
+        old_submission_qs = _use_read_replica(old_submission_qs)
+    try:
+        submission = submission_qs.get(uuid=uuid)
+    except Submission.DoesNotExist:
+        # if this raises a DoesNotExist, assume it's real and let callers deal with it
+        submission = old_submission_qs.get(uuid=uuid)
```

Comment:
Idea: write the new-style hex value back at this point, so future calls don't have to pay the extra cost of the missed first query